### PR TITLE
feat(cross-compile): create tools to cross compile blink and multipass

### PR DESCRIPTION
**Pull request by sdwoodbury**
**What this PR does** 📖

when using tauri mobile to cross compile for Android, some targets had problems with opus being built for arm32 instead of arm64. This branch includes a docker file that fixes this. 

building the docker container
`docker build . -t aarch64-builder -f Dockerfile.aarch64-android`

building apk for uplink (aarch64)
check out this branch in uplink: https://github.com/Satellite-im/Uplink/tree/jk/ios-builds
start the docker container:  ```docker run --rm -ti -v `pwd`:/root/build aarch64-builder```
`cd mobile`
`cargo android apk build aarch64`

testing that warp builds
```docker run --rm -ti -v `pwd`:/root/build aarch64-builder```
`cd extensions/warp-blink-wrtc`
`cargo build --target aarch64-linux-android`

note that the `audio-codec-cli` extension fails to build due to linking errors. tauri-mobile (the `cargo android` tool) doesn't have this problem. 

**Which issue(s) this PR fixes** 🔨
https://github.com/Satellite-im/Uplink/issues/969

**Special notes for reviewers** 🗒️

**Additional comments** 🎤
related to https://github.com/Satellite-im/Uplink/pull/958

### CODE REVIEW

**Comment by sdwoodbury** [comment]
tauri mobile needs the Android SDK, which can be added to a docker container like this: https://github.com/docker-android-sdk/android-30/blob/master/Dockerfile.

File: Dockerfile.aarch64-android [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+WORKDIR /root/rust 
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y
+RUN . $HOME/.cargo/env && rustup target add aarch64-unknown-linux-gnu 
+RUN . $HOME/.cargo/env && rustup toolchain install stable-aarch64-unknown-linux-gnu
+
+WORKDIR /root/cmake
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1.tar.gz
+RUN     tar -xzvf cmake-3.23.1.tar.gz
+RUN cd cmake-3.23.1 && ./configure
+RUN cd cmake-3.23.1 && make -j12
```
Comment (dariusc93) [inline Dockerfile.aarch64-android:19]: maybe the job count should be removed?

File: Dockerfile.aarch64-android [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+RUN cd cmake-3.23.1 && make install
+
+RUN apt install -y protobuf-compiler
```
Comment (dariusc93) [inline Dockerfile.aarch64-android:22]: this is no longer needed as of https://github.com/Satellite-im/Warp/pull/281

```suggestion
```

File: Dockerfile.aarch64-android [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+RUN apt install -y libgtk-3-dev libwebkit2gtk-4.1-dev libayatana-appindicator3-dev librsvg2-dev librust-alsa-sys-dev
+
+# todo: don't need workdir to be /root/rust
+WORKDIR /root/rust 
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y
+RUN . $HOME/.cargo/env && rustup target add aarch64-unknown-linux-gnu 
+RUN . $HOME/.cargo/env && rustup toolchain install stable-aarch64-unknown-linux-gnu
+
+WORKDIR /root/cmake
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1.tar.gz
```
Comment (dariusc93) [inline Dockerfile.aarch64-android:16]: I take it that its easier to download directly vs apt or is this a specific version that you need?

Comment (sdwoodbury) [reply Dockerfile.aarch64-android:16]: In the past I had problems with the version available via `apt` being too low. I did it this way to be safe.

File: Dockerfile.aarch64-android (hunk 1 above)
Comment (sdwoodbury) [reply Dockerfile.aarch64-android]: I replaced it with `nproc --ignore=2` to make it a minimum of 1 and maximum of (number of processors - 2)

File: Dockerfile.aarch64-android (hunk 2 above)
Comment (sdwoodbury) [reply Dockerfile.aarch64-android]: I removed it. also consolidated the `apt install` commands

**Pull request merged**